CMSgov/mint-app
# [EASI-2152] Front End for Ready for Review implementation

# EASI-2152

Figma: https://www.figma.com/file/M72DReE73SSXMN48WZfaKN/MINT---New-Model-Process-(Model-Intake)?node-id=2925%3A219994

## Changes and Description

- Adds a section to end of each task list item for user to indicate that the task list is ready for review
- Change 2



## How to test this change

1. Navigate to the end of each task list
2. New model status box will appear on the last page of each model plan section. 
3. Check box label will read, "This section of the Model Plan (Section name) is ready for review."
![image](https://user-images.githubusercontent.com/2349702/181301395-bfd9b0e5-a1f7-4dbc-b582-b046938caa78.png)

5. Checking this box updates the status on the task list to "Ready for review"
![image](https://user-images.githubusercontent.com/2349702/181301478-0a943819-7000-4fb3-bf1a-7cb95d120fb6.png)

7. Navigate to the last page in the same task list item to see that the checkbox is checked and User can see when it was marked for review and who marked it.  This date should always be the date this box is checked even if other changes are made to the form after it's checked.
![image](https://user-images.githubusercontent.com/2349702/181301556-1e66f6e9-e707-4b33-8046-46634813a44c.png)

9. Unchecking this box will return the status on the task list to "In progress" but the copy should still be there
![image](https://user-images.githubusercontent.com/2349702/181301599-74beda24-880e-484d-9c94-a4673b8854b9.png)
![image](https://user-images.githubusercontent.com/2349702/181301630-07692e94-2055-4e6e-af1f-cc0a58775793.png)


11. Finally, navigate through it one more time to see check status box after it's been unchecked, the line for who originally checked it is gone. 
![image](https://user-images.githubusercontent.com/2349702/181301689-57ef4039-d161-4ac2-a429-8a83a81b739b.png)



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
